--- a/setup.py
+++ b/setup.py
@@ -25,14 +25,8 @@
 if platform.system() in ['Linux', 'FreeBSD', 'DragonFly']:
     parser = argparse.ArgumentParser(add_help=False)
     parser.add_argument('--root=', dest='root_path', metavar='dir', default='/')
-    parser.add_argument('--prefix=', dest='prefix_path', metavar='prefix', default=sys.prefix)
     opts, _ = parser.parse_known_args(sys.argv[1:])
-    usr_share = os.path.join(opts.root_path + opts.prefix_path, "share")
-    if not os.access(usr_share, os.W_OK):
-        if 'XDG_DATA_HOME' in os.environ.keys():
-            usr_share = os.environ['XDG_DATA_HOME']
-        else:
-            usr_share = os.path.expanduser('~/.local/share')
+    usr_share = os.path.join(sys.prefix, "share")
     data_files += [
         # Menu icon
         (os.path.join(usr_share, 'icons/hicolor/128x128/apps/'), ['icons/electron-cash.png']),
